stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/actions/Init.java

```diff
@@ -0,0 +1,123 @@
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.nativeimpl.actions.ClientConnectorFuture;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaAction;
+import org.ballerinalang.net.grpc.ClientConnectorFactory;
+import org.ballerinalang.net.grpc.exception.GrpcClientException;
+import org.ballerinalang.net.grpc.stubs.GrpcBlockingStub;
+import org.ballerinalang.net.grpc.stubs.GrpcNonBlockingStub;
+import org.ballerinalang.net.grpc.stubs.ProtoFileDefinition;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@code Init} is the Init action implementation of the gRPC Connector.
+ */
+@BallerinaAction(
+        packageName = "ballerina.net.grpc",
+        actionName = "<init>",
+        connectorName = "GRPCConnector",
+        args = {@Argument(name = "c", type = TypeKind.CONNECTOR)},
+        connectorArgs = {
+                @Argument(name = "host", type = TypeKind.STRING),
+                @Argument(name = "port", type = TypeKind.INT),
+                @Argument(name = "type", type = TypeKind.STRING),
+                @Argument(name = "descriptorKey", type = TypeKind.STRING),
+                @Argument(name = "descriptorMap", type = TypeKind.MAP)
+        })
+public class Init extends AbstractNativeAction {
+    
+    @Override
+    public ConnectorFuture execute(Context context) {
+        BConnector bConnector = (BConnector) getRefArgument(context, 0);
+        int port = (int) bConnector.getIntField(0);
+        String host = bConnector.getStringField(0);
+        String subtype = bConnector.getStringField(1);
+        //EnumNode.Enumerator enumerator = (EnumNode.Enumerator) bConnector.getRefField(1);
+        String descriptorKey = bConnector.getStringField(2);
+        BMap<String, BValue> descriptorMap = (BMap<String, BValue>) bConnector.getRefField(0);
+        ClientConnectorFuture future = new ClientConnectorFuture();
+        
+        try {
+            // If there are more than one descriptors exist, other descriptors are considered as dependent
+            // descriptors.  client supported only one depth descriptor dependency.
+            List<byte[]> depDescriptorData = new ArrayList<>();
+            byte[] descriptorValue = null;
+            for (String key : descriptorMap.keySet()) {
+                if (descriptorMap.get(key) == null) {
+                    continue;
+                }
+                if (descriptorKey.equals(key)) {
+                    descriptorValue = hexStringToByteArray(descriptorMap.get(key).stringValue());
+                } else {
+                    depDescriptorData.add(hexStringToByteArray(descriptorMap.get(key).stringValue()));
+                }
+            }
+            
+            if (descriptorValue == null) {
+                throw new RuntimeException("Error while establishing the connection. service descriptor is null.");
```

Comment:
For unrecoverable cases we use BallerinaException in connector actions in Ballerina.